### PR TITLE
WDLize GvsPrepareCallset (briefly known as CreateCohortTable)

### PR DESCRIPTION
**Pull request by mmorgantaylor**
specops issue #273: https://github.com/broadinstitute/dsp-spec-ops/issues/273

- renamed `ngs_cohort_extract.py` -> `create_cohort_extract_data_table.py`
- run the script in a WDL (GvsPrepareCallset.wdl)
- use a custom docker - include script for creating and pushing this docker to gcr.io
- enable running as a SA - this has been tested in Terra and works as expected. if using a dataset that requires SA access and the user does not provide a working SA key, they get this error: `User does not have bigquery.jobs.create permission in project specops-variantstore-sa-tests.`

### CODE REVIEW

File: .dockstore.yml [hunk 1]
```diff
@@ -48,6 +48,16 @@ workflows:
...
+   - name: CreateCohortTable
+     subclass: WDL
+     primaryDescriptorPath: /scripts/variantstore/wdl/CreateCohortTable.wdl
+     testParameterFiles:
+       - /scripts/variantstore/wdl/CreateCohortTable.example.inputs.json
+     filters:
+       branches:
+         - master
+         - ah_var_store
+         - mmt_ngs_cohort_extract_wdl
```
Comment (mmorgantaylor) [inline .dockstore.yml:60]: remove before merging

File: scripts/variantstore/wdl/CreateCohortTable.wdl [hunk 2]
```diff
@@ -68,7 +72,8 @@ task CreateCohortTableTask {
             --destination_table ~{destination_cohort_table_name_final} \
             --fq_cohort_sample_names ~{fq_cohort_sample_table_final} \
             --query_project ~{query_project_final} \
-            --fq_sample_mapping_table ~{fq_sample_mapping_table_final}
+            --fq_sample_mapping_table ~{fq_sample_mapping_table_final} \
+            $SA_ARGS
```
Comment (RoriCremer) [inline scripts/variantstore/wdl/CreateCohortTable.wdl:76]: nice

File: scripts/variantstore/wdl/CreateCohortTable.wdl [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        String? query_project
+        String? destination_project
+        String? destination_dataset
+
+        String? destination_cohort_table_name
+        String? fq_cohort_sample_table
+        String? fq_sample_mapping_table
+
+        File? service_account_json
```
Comment (RoriCremer) [inline scripts/variantstore/wdl/CreateCohortTable.wdl:42]: is this still defined as a File even if it's just a path to a json in GCP?

**Review by RoriCremer — APPROVED** [review]

File: scripts/variantstore/wdl/CreateCohortTable.wdl [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    String fq_sample_mapping_table_final = if defined(fq_sample_mapping_table) then "${fq_sample_mapping_table}" else "${project}.${dataset}.sample_info"
+
+    String has_service_account_file = if (defined(service_account_json)) then 'true' else 'false'
+
+    command <<<
+        set -e
+
+        if [ ~{has_service_account_file} = 'true' ]; then
+          SA_FILENAME="sa_key.json"
+          gsutil cp "~{service_account_json}" $SA_FILENAME
```
Comment (kcibul) [inline scripts/variantstore/wdl/CreateCohortTable.wdl:62]: If this is declared as a`File` above, won't it already be copied down to the VM now and the path be a local path?

File: scripts/variantstore/wdl/extract/build_docker.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
+if [ $# -lt 1 ]; then
+    echo "USAGE: ./build_docker.sh [DOCKER_TAG_STRING]"
```
Comment (kcibul) [inline scripts/variantstore/wdl/extract/build_docker.sh:2]: ❤️

File: scripts/variantstore/wdl/CreateCohortTable.wdl [hunk 6]
```diff
@@ -2,8 +2,17 @@ version 1.0
 
 workflow CreateCohortTable {
    input {
-        String project
+        String data_project
         String dataset
```
Comment (kcibul) [inline scripts/variantstore/wdl/CreateCohortTable.wdl:6]: maybe default_dataset for consistency?  wdyt?

File: scripts/variantstore/wdl/CreateCohortTable.wdl [hunk 7]
```diff
@@ -2,8 +2,17 @@ version 1.0
...
+        String fq_cohort_sample_table = "${data_project}.${dataset}.sample_info"
+        String fq_sample_mapping_table = "${data_project}.${dataset}.sample_info"
+
+        String destination_cohort_table_name
+
+        # TODO testing remove this ?
+        String? query_project = data_project
```
Comment (kcibul) [inline scripts/variantstore/wdl/CreateCohortTable.wdl:13]: Per slack convo -- agree with this (and remove all the way down)

File: scripts/variantstore/wdl/extract/Dockerfile [hunk 8]
```diff
@@ -1,14 +1,15 @@
-FROM python:3.7
+FROM gcr.io/google.com/cloudsdktool/cloud-sdk:305.0.0
 
 # Copy the application's requirements.txt and run pip to install
 ADD requirements.txt /app/requirements.txt
 RUN pip install -r /app/requirements.txt
 
 # Add the application source code.
 ADD raw_array_cohort_extract.py /app
```
Comment (kcibul) [inline scripts/variantstore/wdl/extract/Dockerfile:8]: can you remove this from Dockerfile, it's going away soon (and we don't use it).  You can leave the `raw_array_cohort_extract.py` file still, we'll get that with the arrays cleanup

File: scripts/variantstore/wdl/extract/run_gvs_tieout_extract.sh [hunk 9]
```diff
@@ -1,7 +1,7 @@
 PROJECT="spec-ops-aou"
 DATASET="gvs_tieout_acmg_v1"
 
-python ngs_cohort_extract.py \
+python create_cohort_data_table.py \
```
Comment (kcibul) [inline scripts/variantstore/wdl/extract/run_gvs_tieout_extract.sh:4]: suggestion to name this `create_cohort_extract_data_table.py` to be super clear

**Review by kcibul — APPROVED** [review]

File: scripts/variantstore/wdl/CreateCohortTable.wdl (hunk 3 above)
Comment (mmorgantaylor) [reply scripts/variantstore/wdl/CreateCohortTable.wdl:64]: as Kris pointed out, _since_ it's a file, it'll localize, so we _can_ just pass it straight into python. tested!

File: scripts/variantstore/wdl/CreateCohortTable.wdl (hunk 6 above)
Comment (mmorgantaylor) [reply scripts/variantstore/wdl/CreateCohortTable.wdl:6]: nice, yeah, will change

File: scripts/variantstore/wdl/CreateCohortTable.wdl [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        String fq_cohort_sample_table = "~{data_project}.~{default_dataset}.sample_info"
+        String fq_sample_mapping_table = "~{data_project}.~{default_dataset}.sample_info"
+        String fq_temp_table_dataset = "~{destination_project}.temp_tables"
+        String fq_destination_dataset = "~{destination_project}.~{destination_dataset}"
+
+        String? docker
+    }
+
+    # TODO update this docker source
+    String docker_final = select_first([docker, "us.gcr.io/broad-dsde-methods/variantstore:latest"])
```
Comment (mmorgantaylor) [inline scripts/variantstore/wdl/CreateCohortTable.wdl:24]: the `latest` tag doesn't automatically work - need help figuring out what to do here

File: scripts/variantstore/wdl/extract/Dockerfile (hunk 8 above)
Comment (mmorgantaylor) [reply scripts/variantstore/wdl/extract/Dockerfile:8]: ah super, sure

File: scripts/variantstore/wdl/extract/run_gvs_tieout_extract.sh (hunk 9 above)
Comment (mmorgantaylor) [reply scripts/variantstore/wdl/extract/run_gvs_tieout_extract.sh:4]: 👍

**Pull request merged**